### PR TITLE
DPDK: add nff-go test case

### PR DESCRIPTION
**Pull request by mcgov**
- add test suite tool
- add test case in dpdksuite

### CODE REVIEW

**Comment by mcgov** [comment]
This code is broken curently since update_env doesn't support variable expansion.

**Comment by mcgov** [comment]
`not found command: Command not found: sh.`

File: microsoft/testsuites/dpdk/dpdknffgo.py
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdknffgo.py:16]: Can we create a tool for go?

Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdknffgo.py:16]: We can but I'll do it in a new PR once this is merged

**Comment by mcgov** [comment]
Adding updates for a workaround after talking to chi, hold this PR until those are tested

File: microsoft/testsuites/dpdk/dpdksuite.py
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdksuite.py]: The `pass` doesn't need, if there is content in a method.

File: microsoft/testsuites/dpdk/dpdknffgo.py
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdknffgo.py]: Is it useful? We add it in each command already.

File: microsoft/testsuites/dpdk/dpdknffgo.py
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdknffgo.py]: The PATH appears many times, and it's worth to use a variable.

File: microsoft/testsuites/dpdk/dpdksuite.py
Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdksuite.py]: oop u right, will fix

File: microsoft/testsuites/dpdk/dpdknffgo.py
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdknffgo.py]: give it a more meaningful name like `original_path`?

File: microsoft/testsuites/dpdk/dpdknffgo.py
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdknffgo.py]: It doesn't need to save a dependent tool to object level. Even it needs to be used in another method, it can be got again. It doesn't impact the performance, as the tool is reused.

File: microsoft/testsuites/dpdk/dpdknffgo.py
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdknffgo.py]: 1. the install will be called by LISA, it doesn't need to call it.
2. The gcc is a dependency of Make, so it should be installed already.

File: microsoft/testsuites/dpdk/dpdksuite.py
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdksuite.py]: The new priority should be 3. Add 1 based on LISAv2 test case.

File: microsoft/testsuites/dpdk/dpdknffgo.py
Comment (squirrelsc) [reply microsoft/testsuites/dpdk/dpdknffgo.py]: Is it changed? It can be something like `self.update_path = f"{self.original_path}:/usr/local/go/bin"`. With this, the original_path is not needed.

Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdknffgo.py]: Should be fixed

Comment (squirrelsc) [reply microsoft/testsuites/dpdk/dpdknffgo.py]: Is it uploaded?

Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdknffgo.py]: I think yes? This comment window is outdated, if you check the files changed bar it shows the current version

Comment (squirrelsc) [reply microsoft/testsuites/dpdk/dpdknffgo.py]: This comment asks to have a variable like below, so the below string doesn't need to write three times in code. But it's still had three times.

`self.updated_path = f"{self.original_path}:/usr/local/go/bin/"`

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**
